hydroflow_lang/src/graph/flat_to_partitioned.rs

```diff
@@ -612,13 +628,36 @@ impl TryFrom<FlatGraph> for PartitionedGraph {
 /// Returns the ID of X & ID of edge OUT of X.
 fn insert_intermediate_node(
     nodes: &mut SlotMap<GraphNodeId, Node>,
+    operator_instances: &mut SecondaryMap<GraphNodeId, OperatorInstance>,
     ports: &mut SecondaryMap<GraphEdgeId, (PortIndexValue, PortIndexValue)>,
     graph: &mut DiMulGraph<GraphNodeId, GraphEdgeId>,
     node: Node,
     edge_id: GraphEdgeId,
 ) -> (GraphNodeId, GraphEdgeId) {
     let span = Some(node.span());
-    let node_id = nodes.insert(node);
+    let node_id = nodes.insert_with_key(|node_id| {
```

Comment:
would be nicer to choose a different variable name either on the lhs of `let` or here, rather than make the reader think about scoping of `node_id`.

Reply:
Ok here since they are actually the exact same value

Reply:
Reordered it to keep just the `insert` without the closure